src/v/pandaproxy/schema_registry/test/test_json_schema.cc

```diff
@@ -555,7 +555,14 @@ static constexpr auto compatibility_test_cases = std::to_array<
     .reader_schema
     = R"({"type": "array", "additionalItems": {"type": "integer"}, "items": [{"type":"boolean"}]})",
     .writer_schema
-    = R"({"type": "array", "additionalItems": {"type": "integer"}, "items": [{"type":"boolean"}, {"type": "boolean"}]})",
+    = R"({"type": "array", "additionalItems": false, "items": [{"type":"boolean"}, {"type": "number"}]})",
```

Comment:
q: does this behave the same way if we omit "additionalItems"?

Reply:
no, additionalItems default to true if omitted, and true accepts every JSON. it would fail the check with the excess schemas 
`is_superset({boolean}, true)` and it would fail the check with `older.additionalItems` `is_superset({integer}, true)`